### DISCUSSION TITLE
WECONNECT_INTERVAL below 300 is possible?

### DISCUSSION THREAD

**Discussion by MattiJarvinen** [tillsteinbach/VWsFriend#57] [Q&A] [answered]
Since 1 minute is too often... it will fail just above one hour.

What kind of intervals people are using? I'd like to have more accurate charging data for ID.4 of course I could just binary search for a working interval but it takes a lot of time. If someone has a nice working setup with a WECONNECT_INTERVAL below 300 I'd like to hear about it.

**Comment by tillsteinbach** [comment, marked as answer]
I hardcoded the lower limit to 300 seconds to prevent that someone is flooding the VW servers with requests and ruins it for everyone  by pushing VW to close the API. There is not really a technical reason for the 300 seconds.
I'm thinking of maybe changing the interval dynamically based on the condition the car is in.
Regarding your desire to have more accurate data: Do you think of the charging curve? Because the start and end should be already pretty accurate.

**Reply by MattiJarvinen** [reply]
It's pretty darn close in charging and I'm not sure if it can be made any better:
- AC 12.3kW 73 min vs. actual 13.1kW 72 min.
- DC 22.3 kWh 14min 52s vs. actual 21.714kW 15 min 05s

My thoughts were mainly based on IDDatalogger which had even with 1min polling more difference than this, maybe it's because the way it calculated integrals. Then again it might also be how the charge start and end fit into polling interval.

Anyhow, getting more detailed charging curve in DC charging would be of interest but in no way is a priority nor a problem if charging reports stay as accurate as it seems now.

**Reply by tillsteinbach** [reply]
Actually it is not even using an integral. I ask the user to provide the battery capacity and use the SoC to calculate what went in.
So there are some inaccuracies by design. Most importantly you cannot compare the value coming from the charger with what I calculate, because I have no idea what the charging overhead for the specific car is.
My idea for that is that I allow the user to put in the real value in the future and learn from that what I need to add to compensate for that error.
Unfortunately I heard from another user that the SoC in the IDs is not linear. So 1% in the lower SoC is not the same amount of energy than in the higher SoC. To compensate for that I would need to put in curves for each model. But probably that will change by updates...
Maybe it is the best to just live with the 5-10 percent inaccuracies and allow for those who want to track the real values to enter those through the UI.

**Reply by tillsteinbach** [reply]
One question: all DC chargers here around are broken so I was not able to test if AC/DC charging is correctly detected. Does it work for you?

**Reply by MattiJarvinen** [reply]
Seems like it. What are you using for detection?

**Reply by tillsteinbach** [reply]
Not very sophisticated, if charging power was at any point during charging above 11kW it is counted as DC.
